stdlib/public/core/CocoaArray.swift

```diff
@@ -145,9 +145,8 @@ internal struct _CocoaArrayWrapper: RandomAccessCollection {
 
       // Make another pass to retain the copied objects
       var result = unsafe target
-      for _ in bounds {
-        unsafe result.initialize(to: result.pointee)
-        unsafe result += 1
```

Comment:
Removing this increment introduces a bug. This doc-comment applies:
https://github.com/swiftlang/swift/blob/652402e9e83d2e970de5a12c34e419f95d1cb7f3/stdlib/public/core/ArrayBuffer.swift#L314-L316

Reply:
Inserting that doc-comment before line 133 would be a fine change; it's a useful explainer.